src/page-multi-facility/ScenarioLibraryModal.tsx

```diff
@@ -316,6 +372,47 @@ const ScenarioLibraryModal: React.FC<Props> = ({ trigger }) => {
             </DeleteModalContents>
           </ModalDialog>
         </ScenarioLibrary>
+
+        {sharedScenarios?.data?.length ? <h3>Shared Scenarios</h3> : null}
+        <ScenarioLibrary>
```

Comment:
This `<ScenarioLibrary>` library section is pretty repetitive with the `<ScenarioLibrary>` found above.  Is there a way we can refactor this code to reduce the duplication?

Reply:
updated!

Reply:
I see that we got rid of the duplication in this `ScenarioLibrary` which is great, but I think the ` ind === 0` toggles make things a bit challenging to read because it can be difficult for people reading the code to extract meaning from having to know the significance of a specific index position.

I think keeping the scenarios in separate `owned` and `shared` arrays and looping through them individually like you had before was easier to read.  Admittedly, my comment wasn't very guiding but what I was hoping would be possible would be to extract some of the duplicated markup out into its own component (internal to this file).  Then we'd pass the arrays separately into the new component to generate the correct HTML output.  The tricky bit would be handling the subtle differences like showing the delete option in the dropdown.

If this proves to be too challenging, then I think its ok to leave things as they are.

Reply:
Since we'd like to get a release out today, I'd say maybe just timebox this to 30-60 minutes.  If it's taking longer than that, we can merge and address later